dbm-ui/backend/db_services/plugin/nameservice/clb.py

```diff
@@ -17,26 +17,123 @@
-from backend.db_meta.models import Cluster
+from backend.db_meta.models import Cluster, ClusterEntry
+from backend.env import CLB_DOMAIN
+from backend.flow.utils import dns_manage
 
 
-def create_lb_and_register_target(cluster_id: int, creator: str) -> Dict[str, Any]:
-    """创建clb并绑定后端主机"""
+@transaction.atomic
+def tendis_add_clb_domain_4_scene(immute_domain: str, bk_cloud_id: int, created_by: str):
+    """ 增加CLB 域名 """
+    cluster = Cluster.objects.get(bk_cloud_id=bk_cloud_id, immute_domain=immute_domain)
+    clb = cluster.clusterentry_set.filter(cluster_entry_type=ClusterEntryType.CLB.value).first()
+    cluster_entry = ClusterEntry.objects.create(
+        cluster=cluster,
+        cluster_entry_type=ClusterEntryType.CLBDNS,
+        entry="clb.{}".format(cluster.immute_domain),
+        creator=created_by,
+        forward_to_id=clb.id,
+    )
+    cluster_entry.save()
+
+
+@transaction.atomic
+def tendis_bind_clb_domain_4_scene(immute_domain: str, bk_cloud_id: int, created_by: str):
+    """ 主域名直接指向CLB """
+    cluster = Cluster.objects.get(bk_cloud_id=bk_cloud_id, immute_domain=immute_domain)
+    immuteEntry = cluster.clusterentry_set.filter(
+        cluster_entry_type=ClusterEntryType.DNS.value, entry=cluster.immute_domain
+    ).first()
+    clbEntry = cluster.clusterentry_set.filter(cluster_entry_type=ClusterEntryType.CLB.value).first()
+    immuteEntry.forward_to_id = clbEntry.id
+    immuteEntry.creator = created_by
+    immuteEntry.save(update_fields=["forward_to_id", "creator"])
+
+
+@transaction.atomic
+def tendis_unBind_clb_domain_4_scene(immute_domain: str, bk_cloud_id: int, created_by: str):
+    """ 主域名解绑CLB """
+    cluster = Cluster.objects.get(bk_cloud_id=bk_cloud_id, immute_domain=immute_domain)
+    immuteEntry = cluster.clusterentry_set.filter(
+        cluster_entry_type=ClusterEntryType.DNS.value, entry=cluster.immute_domain
+    ).first()
+    immuteEntry.forward_to_id = None
+    immuteEntry.creator = created_by
+    immuteEntry.save(update_fields=["forward_to_id", "creator"])
+
+
+@transaction.atomic
+def delete_clb(domain: str):
+    """删除db中clb数据"""
+
+    cluster = Cluster.objects.filter(immute_domain=domain).get()
+    cluster.clusterentry_set.filter(cluster_entry_type=ClusterEntryType.CLB).delete()
+
+
+@transaction.atomic
+def delete_clb_dns(domain: str):
+    """删除db中clbDns"""
+
+    cluster = Cluster.objects.filter(immute_domain=domain).get()
+    cluster.clusterentry_set.filter(cluster_entry_type=ClusterEntryType.CLBDNS).delete()
+
+
+def tendis_domain_bind_clb_status(immute_domain: str, bk_cloud_id: int) -> bool:
+    """判断主域名是否绑定了clb ip"""
+
+    cluster = Cluster.objects.get(bk_cloud_id=bk_cloud_id, immute_domain=immute_domain)
+    immuteEntry = cluster.clusterentry_set.filter(
+        cluster_entry_type=ClusterEntryType.DNS.value, entry=cluster.immute_domain
+    ).first()
+    if immuteEntry.forward_to_id is None:
+        return False
+    return True
```

Comment:
if immuteEntry.forward_to_id is None:
        return False
    return True

return immute_enntry.forward_to_id is not None
----------------